internal/service/ec2/transitgateway_shared_owners.go

```diff
@@ -0,0 +1,172 @@
+package ec2
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceTransitGatewaySharedOwners() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: ResourceTransitGatewaySharedOwnersCreate,
+		ReadWithoutTimeout:   ResourceTransitGatewaySharedOwnersRead,
+		DeleteWithoutTimeout: ResourceTransitGatewaySharedOwnersDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTransitGatewaySharedOwnersImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"transit_gateway_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func ResourceTransitGatewaySharedOwnersCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).EC2Conn
+
+	transitGatewayID := d.Get("transit_gateway_id").(string)
+	accountID := d.Get("account_id").(string)
+
+	transitGateway, err := DescribeTransitGateway(conn, transitGatewayID)
+	if err != nil {
+		return diag.Errorf("error describing Transit Gateway (%s): %s", transitGatewayID, err)
+	}
+
+	if transitGateway == nil {
+		return diag.Errorf("Transit Gateway %s not found", transitGatewayID)
+	}
+
+	modifyInput := &ec2.ModifyTransitGatewayInput{
+		TransitGatewayId: aws.String(transitGatewayID),
+		Options: &ec2.ModifyTransitGatewayOptions{
+			AddSharedOwners: []*string{aws.String(accountID)},
+		},
+	}
+
+	_, err = conn.ModifyTransitGateway(modifyInput)
+	if err != nil {
+		return diag.Errorf("[ERROR] Error modifying Transit Gateway (%s) to add shared owner (%s): %s", transitGatewayID, accountID, err)
```

Comment:
```suggestion
		return diag.Errorf("error sharing Transit Gateway (%s) to project (%s): %w", transitGatewayID, accountID, err)
```

Дополнительное указание [ERROR] не нужно, оно в логе и так будет. Можно переделать немного текст, чтобы было понятно что по сути просиходит. Также для ошибок используется форматтер `%w`.

Reply:
> Также для ошибок используется форматтер `%w`.

`github.com/hashicorp/terraform-plugin-sdk/v2/diag.Errorf does not support error-wrapping directive %w`